core/lib/multivm/src/versions/vm_fast/bootloader_state/utils.rs

```diff
@@ -107,7 +110,12 @@ pub(crate) fn apply_l2_block(
         ),
         (
             block_position + 3,
-            bootloader_l2_block.max_virtual_blocks_to_create.into(),
+            if start_new_l2_block {
+                bootloader_l2_block.max_virtual_blocks_to_create
+            } else {
+                0
+            }
+            .into(),
```

Comment:
Bikeshedding: IMO, moving `.into()` to the `if` clause and using `U256::zero()` would be more readable.